test/util/test_case.cpp

```diff
@@ -19,25 +19,50 @@
 #include "gtest/gtest.h"
 #include "ngraph/assertion.hpp"
 
+ngraph::test::NgraphTestCase::NgraphTestCase(const std::shared_ptr<Function>& function,
+                                             const std::string& backend_name,
+                                             const BackendMode mode)
+    : m_function(function)
+    , m_backend(ngraph::runtime::Backend::create(backend_name,
+                                                 mode == BackendMode::DYNAMIC ? true : false))
+{
+    if (mode == BackendMode::STATIC)
+    {
+        NGRAPH_CHECK(!m_function->is_dynamic(),
+                     "For dynamic function using dynamic backend is expected.");
+    }
+    m_executable = m_backend->compile(m_function);
+    for (auto i = 0; i < m_function->get_output_size(); ++i)
+    {
+        const auto& output_tensor =
+            (mode == BackendMode::DYNAMIC)
+                ? m_backend->create_dynamic_tensor(m_function->get_output_element_type(i),
+                                                   m_function->get_output_partial_shape(i))
+                : m_backend->create_tensor(m_function->get_output_element_type(i),
+                                           m_function->get_output_shape(i));
+
+        m_result_tensors.emplace_back(output_tensor);
+    }
+}
+
 void ngraph::test::NgraphTestCase::run(size_t tolerance_bits)
 {
     m_tolerance_bits = tolerance_bits;
     const auto& function_results = m_function->get_results();
     NGRAPH_CHECK(m_expected_outputs.size() == function_results.size(),
                  "Expected number of outputs is different from the function's number of results.");
-
-    auto handle = m_backend->compile(m_function);
-    handle->call_with_validate(m_result_tensors, m_input_tensors);
+    m_executable->call_with_validate(m_result_tensors, m_input_tensors);
 
     for (size_t i = 0; i < m_expected_outputs.size(); ++i)
     {
         const auto& result_tensor = m_result_tensors.at(i);
         const auto& expected_result_constant = m_expected_outputs.at(i);
         const auto& element_type = result_tensor->get_element_type();
 
-        auto expected_shape = expected_result_constant->get_shape();
-        auto result_shape = result_tensor->get_shape();
-        EXPECT_EQ(expected_shape, result_shape);
+        EXPECT_EQ(expected_result_constant->get_output_size(), 1);
+        const auto& expected_shape = expected_result_constant->get_output_partial_shape(0);
+        const auto& result_shape = result_tensor->get_partial_shape();
+        EXPECT_TRUE(result_shape.compatible(expected_shape));
```

Comment:
I think this check should be more strict and should operate on `Shape`s, not `PartialShape`s. This is because both expected results and actual results caluculated with `m_executable->call_with_validate()` contain only static shapes.

Reply:
@tomdol Thanks, good point